PLT-600 Renamed channel description to channel header and added channel purpose field

hmhealey opened the pull request.

coreyhulen commented on store/sql_channel_store.go:
```diff
@@ -25,7 +25,8 @@ func NewSqlChannelStore(sqlStore *SqlStore) ChannelStore {
 		table.SetUniqueTogether("Name", "TeamId")
-		table.ColMap("Description").SetMaxSize(1024)
+		table.ColMap("Header").SetMaxSize(1024)
+		table.ColMap("Purpose").SetMaxSize(128)
```
why so small?

hmhealey replied:
The ticket only specified 100 characters although I bumped it up to 128 to keep the trend of powers of 2 for the field sizes. I assume that's because purpose is just shown in the More Channels menu so we can only fit a certain amount.

coreyhulen commented on model/channel.go:
```diff
@@ -24,7 +24,8 @@ type Channel struct {
 	Name          string `json:"name"`
-	Description   string `json:"description"`
+	Header        string `json:"header"`
+	Purpose       string `json:"purpose"`
```
I don't think this will work on postgres upgrade.  New columns must be appended.

hmhealey replied:
I'll load up my postgres instance and check that out to see if that's the case

hmhealey commented:
I just updated my local postgres instance from whatever it had been running prior to this change and it worked perfectly so there's no issues with the column ordering or anything.

coreyhulen commented:
Are you sure on the postgres thing?  You should run the unit tests in the 1.1 branch then checkout your branch without a `make clean` and I'm 99% sure the upgrade will fail.

hmhealey commented:
Yep, it works fine. Just dropped everything from my postgres DB, switched to the release-1.1.0 branch, ran tests, switched to this branch, and then ran tests again and it worked fine. I also tested to make sure the channel description field was also properly renamed to channel header and kept its previous value.

coreyhulen commented:
Hmm...something must have changed in gorp then.  Because this shouldn't even work for mysql.  If you look at some of the really old code https://github.com/mattermost/platform/blob/c84baf230ccb6f95fcf43798a3eb837c625639db/store/sql_store.go#L129 We needed the AFTER Column stuff in the create column query to make it work properly with mysql.  Postgres doesn't support the AFTER keyword.

hmhealey commented:
Not sure what would've changed then since we haven't updated gorp since the first commit to this repo. Perhaps it's something in go's sql package since listing the columns in psql even shows them as out of order.
![image](https://cloud.githubusercontent.com/assets/3277310/10772833/51522c62-7cce-11e5-881f-0aa385c6e272.png)

jwilander commented:
+1 if we're sure the DB upgrades will work. Also in the future it might be best to avoid renaming DB columns unless really necessary. For example in this case the `Header` could have kept the name `Description` internally and we could just refer to it as the `Header` in UI text.
